app/src/main/java/com/github/h3lp3rs/h3lp/forum/FireForum.kt

```diff
@@ -4,55 +4,62 @@ import android.os.Build
 import androidx.annotation.RequiresApi
 import com.github.h3lp3rs.h3lp.database.Databases.Companion.databaseOf
 import com.github.h3lp3rs.h3lp.database.Databases.FORUM
-import com.github.h3lp3rs.h3lp.forum.*
-import com.github.h3lp3rs.h3lp.forum.ForumCategory.*
 import com.github.h3lp3rs.h3lp.forum.ForumCategory.Companion.DEFAULT_CATEGORY
 import com.github.h3lp3rs.h3lp.forum.ForumCategory.Companion.forumOf
+import com.github.h3lp3rs.h3lp.forum.ForumCategory.valueOf
+import com.github.h3lp3rs.h3lp.forum.ForumCategory.values
 import com.github.h3lp3rs.h3lp.forum.data.ForumPostData
 import java.time.ZonedDateTime
 import java.util.concurrent.CompletableFuture
 
-class FireForum(override val path: Path) : Forum {
+class FireForum(override val path : Path) : Forum {
     private val rootForum = databaseOf(FORUM)
 
     @RequiresApi(Build.VERSION_CODES.O)
-    override fun newPost(author: String, content: String): CompletableFuture<ForumPost> {
+    override fun newPost(author : String, content : String) : CompletableFuture<ForumPost> {
         var key = ""
         // Incrementing by 2 so that a post's replies key is always 1 more than a post's key (this
         // is also an optimization to avoid us requiring 2 calls to incrementAndGet)
         return rootForum.incrementAndGet(UNIQUE_POST_ID, 2).thenApply { postKey ->
             key = postKey.toString()
             val repliesKey = (postKey + 1).toString()
-
+            val currentTime = ZonedDateTime.now()
             val forumPostData = ForumPostData(
                 author,
                 content,
-                ZonedDateTime.now(),
+                getFormattedPostTime(currentTime),
                 key,
                 repliesKey,
                 getCurrentCategory()
             )
             rootForum.setObject(
-                pathToKey(path + key),
-                ForumPostData::class.java,
-                forumPostData
+                pathToKey(path + key), ForumPostData::class.java, forumPostData
             )
 
             // If the forum is a category, this means that the post is a "main post" (not a reply)
             // we thus add it to the posts in the database (as explained in the forum protocol)
             if (isCategory()) {
                 rootForum.addToObjectsListConcurrently(
-                    pathToKey(path + POSTS_LIST),
-                    String::class.java,
-                    key
+                    pathToKey(path + POSTS_LIST), String::class.java, key
```

Comment:
On a side note, your Android Studios reformatter is different than mine (and the others I think as well).
Thus, a lot of changes are signaled without a valid reason by Git.
It's a bit late to change of course, but having an auto formatter in the CI would be great.